examples/basics/scene/points_selection.py

```diff
@@ -0,0 +1,271 @@
+            self.view.camera._viewbox.events.mouse_move.connect(
+                    self.view.camera.viewbox_mouse_event)
+            self.view.camera._viewbox.events.mouse_press.connect(
+                        self.view.camera.viewbox_mouse_event)
+            self.view.camera._viewbox.events.mouse_release.connect(
+                        self.view.camera.viewbox_mouse_event)
+            self.view.camera._viewbox.events.mouse_wheel.connect(
+                        self.view.camera.viewbox_mouse_event)
+
+    def mark_selected(self):
+        # Change the color of the selected point
+
+        self.facecolor[self.facecolor[:,1] != 1.0] = self.white
+        self.scatter.set_data(self.data, face_color=self.facecolor, size=self.ptsize)
+        for i in self.selected:
+            self.facecolor[i] = [1.0, 0.0, 0.0, 1]
+
+        self.scatter.set_data(self.data, face_color = self.facecolor, size=self.ptsize)
+        self.scatter.update()
+
+    def on_key_press(self, event):
+        # Set selection_flag and instruction text
+
+        if event.text in self.selection_pool.keys():
+            if not self.selection_flag:
+                self.text.text = 'Now is %s selection mode, press %s to switch' % (self.selection_pool[event.text], event.text)
+                self.selection_flag = True
+            else:
+                self.text.text = 'Now is view mode, press %s to switch' % event.text
+                self.selection_flag = False
+            self.event_connect(self.selection_flag)
+            self.selection_id = event.text
+
+    def on_mouse_press(self, event):
+        # Realize picking functionality and set origin mouse pos
+
+        if event.button == 1 and self.selection_flag:
+            if self.selection_id == '4':
+                # Ray intersection on the CPU to highlight the selected point(s)
+
+                data = self.tr.map(self.data)[:, :2]
+                m1 = data > (event.pos - 4)
+                m2 = data < (event.pos + 4)
+
+                self.selected = np.argwhere(m1[:,0] & m1[:,1] & m2[:,0] & m2[:,1])
+                self.mark_selected()
+
+            else:
+                self.selection_origin = event.pos
+
+    def on_mouse_release(self, event):
+        # Identify selected points and mark them
+
+        if event.button == 1 and self.selection_flag and self.selection_id is not '4':
+            self.facecolor[self.facecolor[:,1] != 1.0] = self.white
+
+            data = self.tr.map(self.data)[:,:2]
+
+            if self.selection_id in ['1', '2', '3']:
+                selection_path = path.Path(self.line_pos, closed=True)
+                mask = [selection_path.contains_points(data)]
+
+                self.selected = mask
+                self.mark_selected()
+
+                # Reset lasso
+                self.line_pos = []  # TODO: Empty pos input is not allowed for line_visual
+                self.line.set_data(np.array(self.line_pos))
+                self.line.update()
+
```

Comment:
Here will pop up an `ValueError`  that the `LineVisual` doesn't accept empty input data as pos, is there any other way to do the reset for the `LineVisual` ? Thanks!

The error info:
```
...
File "/Users/penny/anaconda/lib/python2.7/site-packages/vispy-0.5.0.dev0-py2.7.egg/vispy/visuals/line/line.py", line 312, in _prepare_draw
    self._pos_vbo.set_data(pos)
  File "/Users/penny/anaconda/lib/python2.7/site-packages/vispy-0.5.0.dev0-py2.7.egg/vispy/gloo/buffer.py", line 192, in set_data
    data = self._prepare_data(data, **kwargs)
  File "/Users/penny/anaconda/lib/python2.7/site-packages/vispy-0.5.0.dev0-py2.7.egg/vispy/gloo/buffer.py", line 451, in _prepare_data
    % (self._last_dim, c))
ValueError: Last dimension should be 2 not 1
```